### PR TITLE
Parse headers with no whitespaces

### LINKED ISSUE #854: DefaultHttpHeaders does not hold http header properly.
`DefaultHttpHeaders` does not hold http header values as client send.  The http header value from `DefaultHttpHeaders#get(String)` is missing the first character.

---

### servicetalk version

0.19.0

### output of `uname -a`

```
 uname -a
Darwin mike-neck.local 19.0.0 Darwin Kernel Version 19.0.0: Wed Sep 25 20:18:50 PDT 2019; root:xnu-6153.11.26~2/RELEASE_X86_64 x86_64
```

### output of `java -version`

```
$ java -version
Picked up _JAVA_OPTIONS: -Dfile.encoding=UTF-8
openjdk version "1.8.0_202"
OpenJDK Runtime Environment (Zulu 8.36.0.1-CA-macosx) (build 1.8.0_202-b05)
OpenJDK 64-Bit Server VM (Zulu 8.36.0.1-CA-macosx) (build 25.202-b05, mixed mode)
```

### context

* to retrieve http header value

---

Reproduction code
---

### server

```groovy
@GrabResolver(name = 'servicetalk', root='https://dl.bintray.com/servicetalk/servicetalk/')
@Grab("io.servicetalk:servicetalk-http-netty:0.19.0")
@Grab("io.servicetalk:servicetalk-annotations:0.19.0")
@Grab('org.slf4j:slf4j-simple:1.7.29')

import io.servicetalk.concurrent.api.Single
import io.servicetalk.http.api.*
import io.servicetalk.http.netty.HttpServers

import java.util.concurrent.*

import static io.servicetalk.http.api.HttpSerializationProviders.textSerializer


HttpServers.forPort(8000)
    .listenAndAwait([handle: {HttpServiceContext ctx, HttpRequest req, HttpResponseFactory factory ->
        def myHeader = req.headers().get('MY')
        Single.succeeded(factory.ok().payloadBody("${myHeader}" as String, textSerializer()))
    }] as HttpService).awaitShutdown()
```

(groovy code)

### client

```
curl -i http://localhost:8000 -H 'MY:foo-bar'
```

This client expects to receive response as ...

```
HTTP/1.1 200 OK
content-length: 6
content-type: text/plain; charset=UTF-8

foo-bar
```

But it will receive as bellow.

```
HTTP/1.1 200 OK
content-length: 6
content-type: text/plain; charset=UTF-8

oo-bar
```

### PR DESCRIPTION
**Pull request by idelpivnitskiy**
Motivation:

The whitespace between the `:` and a header value is optional [1] but we
consider it as required [2] hence skip a character after the `:`. As a result,
a header value may miss the first character.

[1] https://tools.ietf.org/html/rfc7230#section-3.2
[2] `header-field = field-name ":" OWS field-value OWS`

Modifications:

- Do not skip a character after `:`;
- Add more tests for headers with no whitespace and for mixed
whitespaces;

Result:

Headers parser does not skip the first value character when there is no
whitespace after `:`.

Resolves #854

### CODE REVIEW

File: servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpObjectDecoder.java [hunk 1]
```diff
@@ -544,40 +544,42 @@ private void parseHeaderLine(HttpHeaders headers, ByteBuf buffer, final int lfIn
...
-            int valueEnd;
+            final int valueEnd;
             if (valueStart < 0 || (valueEnd = buffer.forEachByteDesc(valueStart, lfIndex - valueStart - 1,
                     FIND_NON_LINEAR_WHITESPACE)) < 0) {
                 headers.add(name, emptyAsciiString());
             } else {
                 // We assume the allocator will not leak memory, and so we retain + slice to avoid copying data.
-                headers.add(name, newAsciiString(newBufferFrom(
-                        buffer.retainedSlice(valueStart, valueEnd - valueStart + 1))));
+                final CharSequence value = newAsciiString(newBufferFrom(
```
Comment (Scottmitch) [inline servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpObjectDecoder.java]: IIUC this change is formatting related ... can you revert and keep the inlined approach as before?

File: servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/HttpResponseDecoderTest.java [hunk 2]
```diff
@@ -118,6 +118,40 @@ public void contentLengthNoTrailersHeaderWhiteSpace() {
...
+    @Test
+    public void contentLengthNoTrailersHeaderMixedWhiteSpace() {
+        EmbeddedChannel channel = newEmbeddedChannel();
+        byte[] content = new byte[128];
+        ThreadLocalRandom.current().nextBytes(content);
+        byte[] beforeContentBytes = ("HTTP/1.1 200 OK" + "\r\n" +
+                "Connection   :keep-alive" + "\r\n" +
+                "   Server   :unit-test" + "\r\n" +
+                "Empty:" + "\r\n" +
+                "EmptyWhitespace:   " + "\r\n" +
```
Comment (Scottmitch) [inline servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/HttpResponseDecoderTest.java]: consider adding a `SingleCharacterNoWhiteSpace: a` to make sure we don't overrun in this case.
